### PR TITLE
Topic05 react ex1/2

### PR DESCRIPTION
**Pull request by Igrass97**
This covers ex 1 and 2. It's a lot better than the first time

### CODE REVIEW

File: Topic 5 - React JS/Ex 1/ex1/src/App.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      ],
+      currentId: 3
+    }
+
+    this.addMovie = this.addMovie.bind(this);
+    this.incrementId = this.incrementId.bind(this);
+  }
+
+  addMovie(movieDetails){
+    const state = this.state;
```
Comment (DemianMoschin-GLB) [inline Topic 5 - React JS/Ex 1/ex1/src/App.js]: this is not ok
You are mutating state. 
You should do a shallow copy of state, change that copy and then set the new state

File: Topic 5 - React JS/Ex 1/ex1/src/components/MovieAdder.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(e){
+    const target = e.target;
+    const value = target.value;
+    const name = target.name;
+
+    this.setState({
+      [name]: value
```
Comment (DemianMoschin-GLB) [inline Topic 5 - React JS/Ex 1/ex1/src/components/MovieAdder.js]: +1 
this is great!

File: Topic 5 - React JS/Ex 1/ex1/src/App.js (hunk 1 above)
Comment (DemianMoschin-GLB) [reply Topic 5 - React JS/Ex 1/ex1/src/App.js]: I don't think you need to do this. 

Your method should create a movie, and then apply setState.

File: Topic 5 - React JS/Ex 1/ex1/src/components/MovieAdder.js (hunk 2 above)
Comment (Igrass97) [reply Topic 5 - React JS/Ex 1/ex1/src/components/MovieAdder.js]: Followed the docs haha

**Comment by Igrass97** [comment]
I think now is good, I learned how to use the spread operator to push the new movie and I think I'm not mutating the state directly.

File: Topic 5 - React JS/Ex 1/ex1/src/App.js [hunk 3]
```diff
@@ -23,11 +23,12 @@ class App extends Component {
...
 
   addMovie(movieDetails){
-    const state = this.state;
     const newMovie = new Movie(movieDetails.id, movieDetails.title, movieDetails.duration, movieDetails.year);
-    this.state.movies.push(newMovie);
-    this.state.currentId++;
-    this.setState(state);
+
+    this.setState(oldState => ({
+      movies: [...oldState.movies, newMovie],
```
Comment (DemianMoschin-GLB) [inline Topic 5 - React JS/Ex 1/ex1/src/App.js]: +1

**Pull request merged**